### ISSUE TITLE
Speed Ez - Incorrect Billing

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1330]
In GitLab by @kdjstudios on Jan 16, 2019, 08:28

**Submitted by:** Jo Ann Browne <joann@speedez.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/6614402

**Server:** External

**Client/Site:** Speed Ez

**Account:** 9009

**Issue:** 

I know that I can be wrong on many things, but, please help me figure out why the ledger is so WRONG on account 9009?

1.  Let’s begin with a Zero Balance on 5/11/2018

2.  ALL charges beginning with Invoice SPDZ-20240 through Invoice SPDZ-24510 equals = $ $10,446.84

3.  ALL payments through 12/10/18 equals = $6,354.17

4.  $10,446.84 minus $6,354.17 equals = $4,092.67. 

5. In my head the BALANCE line on the 1/1/2029 Invoice should have been $4,092.67 and NOT $2,374.55. That is a difference of $1,718.12. 

I really need to know if this big mess-up is from my office or your office and how will or can it be rectified. I can tell you right now my customer certainly is not going to be happy and will stop using my Service if I have to go to them and explain AND what if this is not the only account affected!!!

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 16, 2019, 10:02

I looked into the account and it would appear that Jo Ann did not include all payments in her calculation. 

I wrote:

> Thank you for reaching out to us. I took a look at the account in question. It would appear you missed calculating in the following payment: $1,718.12 - Check - 30287/SPDZ-21961 paid on 10/10/18. This payment is being after the invoice SPDZ-20240 on 6/1/18, so it should be included in your calculation. 
> 
> Please let us know if that resolves your concern.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 16, 2019, 11:49

Jo Ann wrote:

> I need you to call me because I do not see what you are seeing.  210-316-5455


### After call

I called Jo Ann and we discussed the account ledger. I explained that we are currently working on a sorting issue with the account ledger that may also resolve this concern (#1250). But we also want the team to confirm that this is not another separate issue. Below I will explain what me and Jo Ann were able to discuss.

- It looks like the amount and balance columns are not being calculated correctly. For example please look at the invoices lines on 11-1-18 and 12-1-18.

On the 11-1-18 line with invoice SPDZ-23664 where is the -1484.67 balance coming from? This makes no sense to us. The balance column should be the 233.45 correct? 

On the 12-1-18 line with the invoice SPDZ-24098 the balance should be 1753.99 as (233.45 + 1404.66 + 115.88 = 1753.99).

![image](/uploads/65928d5d17f457ebb9b37a00253368d7/image.png)


In summary the invoices do appear to be correct, so what is going to the client is good, but what is being displayed on the account ledger does not appear to match the invoices. 

### Ability to hide pending items.

Jo Ann had mentioned something about how staged fees are being displayed in the account ledger too and how they can add an extra layer of confusion. I had asked if being able to choose to display/hide those pending items would help with that? She agreed. We may want to discuss this with @tim.traylor to determine if we should open a new ticket for that functionality.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 16, 2019, 11:50

@smasih @amishra @pchaudhary 

Please look into what is causing this and provide an explanation. Please provide an estimate if a fix is needed.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 17, 2019, 01:43

@kdjstudios 

* We looked into this & found that the root cause of this is already fixed on the #1250.  
* We also took the backup on our dev server to check the fixed ledger code & after removing the **pending staged fees** from the account ledger, the account ledger balance is showing the same amount as the account balance.  
  * ![image](/uploads/5f7e7da18f1af25e6872d4a36755a996/image.png)

* However, the remaining thing of pending fees can be discussed with Tim.

Hope this helps.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 17, 2019, 13:49

I wrote:

> We are just following up. The support team was able to look into this and they are expecting to have the Account Ledger calculations updated this Saturday (1-19-19). I will follow up once the update is published to insure it fully resolved the discrepancies. Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 21, 2019, 13:39

@pchaudhary @amishra @smasih @tim.traylor 

After reviewing the account after the updated Account Ledger was released, this appears to have negatively effected the account. The balances are completely different now. The balance shows as 4092.67 at the last finalized invoice. However, the actual account balance shows as 2338.68. 

![image](/uploads/8acea2deb2b51bbd5d37d91b2cf6ec88/image.png)
![image](/uploads/a665bc8667bce60110cc37355fedd892/image.png)


Please look into what was missed when this was tested and checked in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1333#note_32973; as it would appear that what ever was tested was not correct.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 22, 2019, 03:13

Hi [Kyle] ( @kdjstudios ), 

I understand your concern & would like to mention a point that which I also mentioned on my [comment] (http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1333#note_32973):  

* We checked the account ledger details by removing the **pending staged fees** from the code of account ledger, as the pending fees doesn't effect the account balance of an account.

* We also would like to mention that we had checked again by removing the pending fees code from this account on the backend & it's showing correct balance.

Hope this helps.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 22, 2019, 12:16

@pchaudhary 

Thank you for the update. I figured out that I was missing adding(actually removing) the last payment amount on the account. 

I wrote:

> The support team released the update to the account ledger and your account should be displaying correctly now. Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 29, 2019, 12:12

closed
